### PR TITLE
r/rabbitmq_secret_backend - add `password_policy` and `username_template` arguments.

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-vault/blob/master/CHANGELOG.md):

```release-note
resource_rabbitmq_secret_backend - add `password_policy` and `username_template` arguments.
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccRabbitmqSecretBackend_basic'
--- PASS: TestAccRabbitMQSecretBackend_basic (2.28s)
--- PASS: TestAccRabbitMQSecretBackend_template (1.23s)
```

### CODE REVIEW

File: vault/resource_rabbitmq_secret_backend.go [hunk 1]
```diff
@@ -54,20 +53,17 @@ func rabbitMQSecretBackendResource() *schema.Resource {
...
 				Type:        schema.TypeString,
 				Required:    true,
-				ForceNew:    true,
 				Sensitive:   true,
 				Description: "Specifies the RabbitMQ management administrator username",
 			},
 			"password": {
 				Type:        schema.TypeString,
 				Required:    true,
-				ForceNew:    true,
```
Comment (benashz) [inline vault/resource_rabbitmq_secret_backend.go:70]: The changes to `ForceNew` seem to be out of scope. It would be good move them into another PR so that we can get core functionality in.

File: vault/resource_rabbitmq_secret_backend.go [hunk 2]
```diff
@@ -160,7 +168,7 @@ func rabbitMQSecretBackendUpdate(d *schema.ResourceData, meta interface{}) error
 
 	path := d.Id()
 	d.Partial(true)
-	if d.HasChange("default_lease_ttl_seconds") || d.HasChange("max_lease_ttl_seconds") {
+	if d.HasChanges("default_lease_ttl_seconds", "max_lease_ttl_seconds") {
```
Comment (benashz) [inline vault/resource_rabbitmq_secret_backend.go:171]: 👍

File: vault/resource_rabbitmq_secret_backend_test.go [hunk 3]
```diff
@@ -24,58 +25,34 @@ func TestAccRabbitMQSecretBackend_basic(t *testing.T) {
...
 				Config: testAccRabbitMQSecretBackendConfig_basic(path, connectionUri, username, password),
 				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr("vault_rabbitmq_secret_backend.test", "path", path),
-					resource.TestCheckResourceAttr("vault_rabbitmq_secret_backend.test", "description", "test description"),
-					resource.TestCheckResourceAttr("vault_rabbitmq_secret_backend.test", "default_lease_ttl_seconds", "3600"),
-					resource.TestCheckResourceAttr("vault_rabbitmq_secret_backend.test", "max_lease_ttl_seconds", "86400"),
-					resource.TestCheckResourceAttr("vault_rabbitmq_secret_backend.test", "connection_uri", connectionUri),
-					resource.TestCheckResourceAttr("vault_rabbitmq_secret_backend.test", "username", username),
-					resource.TestCheckResourceAttr("vault_rabbitmq_secret_backend.test", "password", password),
+					resource.TestCheckResourceAttr(resourceName, "path", path),
```
Comment (benashz) [inline vault/resource_rabbitmq_secret_backend_test.go:28]: It be good to include tests for the newly added fields.

File: vault/resource_rabbitmq_secret_backend_test.go [hunk 4]
```diff
@@ -24,58 +25,34 @@ func TestAccRabbitMQSecretBackend_basic(t *testing.T) {
...
-					resource.TestCheckResourceAttr("vault_rabbitmq_secret_backend.test", "connection_uri", connectionUri),
-					resource.TestCheckResourceAttr("vault_rabbitmq_secret_backend.test", "username", username),
-					resource.TestCheckResourceAttr("vault_rabbitmq_secret_backend.test", "password", password),
-				),
-			},
-		},
-	})
-}
-
-func TestAccRabbitMQSecretBackend_import(t *testing.T) {
```
Comment (benashz) [inline vault/resource_rabbitmq_secret_backend_test.go:52]: Was there specific reason to consolidate this test into the one above?

Comment (DrFaust92) [reply vault/resource_rabbitmq_secret_backend_test.go:52]: I go by the convention in the AWS provider (and even K8s provider IIRC) where the import check is done on every test case, this way you have better coverage or all argument import tests. I dont see much meaning in a dedicated import test case as it usually covers only the basic config but ill go by what ever guidance you decide here.

File: vault/resource_rabbitmq_secret_backend_test.go (hunk 3 above)
Comment (DrFaust92) [reply vault/resource_rabbitmq_secret_backend_test.go:28]: added a dedicated test (not to mess with basic one)

File: vault/resource_rabbitmq_secret_backend.go (hunk 1 above)
Comment (DrFaust92) [reply vault/resource_rabbitmq_secret_backend.go]: Removed

File: vault/resource_rabbitmq_secret_backend_test.go (hunk 4 above)
Comment (benashz) [reply vault/resource_rabbitmq_secret_backend_test.go:52]: Thanks for the detailed response. For PRs like this one we usually prefer to keep the scope of the changes as narrow as possible, in order to limit the possibility of regressions.

**Review by benashz — APPROVED** [review]
LGTM! Thanks again for your contribution to HashiCorp!
